central/cve/datastoretest/datastore_sac_test.go

```diff
@@ -0,0 +1,1239 @@
+			contextKey:     sacTestUtils.Cluster2NamespacesBCReadWriteCtx,
+			expectedExists: true,
+		},
+		"Wrong cluster full read-write can NOT see CVE from shared components": {
+			contextKey: sacTestUtils.Cluster3ReadWriteCtx,
+		},
+		"Wrong cluster partial read-write to some matching namespace can NOT see CVE from shared components": {
+			contextKey: sacTestUtils.Cluster3NamespaceAReadWriteCtx,
+		},
+		"Wrong cluster partial read-write to some other matching namespace can NOT see CVE from shared components": {
+			contextKey: sacTestUtils.Cluster3NamespaceBReadWriteCtx,
+		},
+		"Wrong cluster partial read-write to any matching namespace can NOT see CVE from shared components": {
+			contextKey: sacTestUtils.Cluster3NamespacesABReadWriteCtx,
+		},
+	}
+	for name, c := range testCases {
+		s.Run(name, func() {
+			if features.PostgresDatastore.Enabled() {
+				s.T().Skip("Skipping CVE tests for postgres for now.")
+			}
+			testCtx := s.imageTestContexts[c.contextKey]
+			exists, err := s.imageCVEStore.Exists(testCtx, cveId)
+			s.NoError(err)
+			s.Equal(c.expectedExists, exists)
+		})
+	}
+}
+
+func (s *cveDataStoreSACTestSuite) TestSACImageCVEGetSingleScopeOnly() {
+	// Inject the fixture graph, and test retrieval for CVE-1234-0001
+	s.dackboxTestStore.PushImageToVulnerabilitiesGraph()
+	defer s.cleanImageToVulnerabilitiesGraph()
+	testCases := map[string]getTestCase{
+		"Unrestricted read-write can see single scope CVE": {
+			contextKey:    sacTestUtils.UnrestrictedReadWriteCtx,
+			expectedFound: true,
+		},
+		"Unrestricted read can see single scope CVE": {
+			contextKey:    sacTestUtils.UnrestrictedReadCtx,
+			expectedFound: true,
+		},
+		"Right cluster full read-write can see single scope CVE": {
+			contextKey:    sacTestUtils.Cluster1ReadWriteCtx,
+			expectedFound: true,
+		},
+		"Right cluster partial read-write to right namespace can see single scope CVE": {
+			contextKey:    sacTestUtils.Cluster1NamespaceAReadWriteCtx,
+			expectedFound: true,
+		},
+		"Right cluster partial read-write to wrong namespace can NOT see single scope CVE": {
+			contextKey: sacTestUtils.Cluster1NamespaceBReadWriteCtx,
+		},
+		"Right cluster partial read-write to at least the right namespace can see single scope CVE": {
+			contextKey:    sacTestUtils.Cluster1NamespacesABReadWriteCtx,
+			expectedFound: true,
+		},
+		"Wrong cluster full read-write can NOT see single scope CVE": {
+			contextKey: sacTestUtils.Cluster2ReadWriteCtx,
+		},
+		"Wrong cluster partial read-write to the matching namespace can NOT see single scope CVE": {
+			contextKey: sacTestUtils.Cluster2NamespaceAReadWriteCtx,
+		},
+		"Wrong cluster partial read-write to any not-matching namespace can NOT see single scope CVE": {
+			contextKey: sacTestUtils.Cluster2NamespacesBCReadWriteCtx,
+		},
+	}
+	cve := "CVE-1234-0001"
+	cveId := s.getCVEID(cve)
+	cvss := float32(5.8)
```

Comment:
Would also be better to use the value here provided from the fixture, in case we ever change it.
Same applies for few other places here.